File: solutions/php/resistor-color-duo/1/ResistorColorDuo.php
<?php
declare(strict_types=1);

class ResistorColorDuo
{
    const COLORS = [
        'black' => 0,
        'brown' => 1,
        'red' => 2,
        'orange' => 3,
        'yellow' => 4,
        'green' => 5,
        'blue' => 6,
        'violet' => 7,
        'grey' => 8,
        'white' => 9,
    ];

    public function getColorsValue(array $colors): int
    {
        return self::COLORS[$colors[0]] * 10 + self::COLORS[$colors[1]];
    }
}

File: solutions/clojure/resistor-color-duo/1/src/resistor_color_duo.clj
(ns resistor-color-duo)

(def colors
  ["black" "brown" "red" "orange" "yellow" "green" "blue" "violet" "grey" "white"]
)

(defn color-code [color]
  (.indexOf colors color)
)

(defn resistor-value [[first-color second-color]]
  (+ (* (color-code first-color) 10)
     (color-code second-color)
  )
)
